### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by forrestfwilliams**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,14 +1,35 @@
 # HyP3-lib
 
-## Common library for HyP3 plugins
+## CURRENTLY UNDER CONSTRUCTION!!!
+HyP3-lib is currently undergoing a major redesign and many utilities that are no longer used by ASF have been removed between version 2 and version 3. In addition, we plan to migrate some remaining functionality to other projects and to significantly refactor what remains. For this reason, **if you are using HyP3-lib we strongly recommend pinning to major versions at this time**. We further recommend not using this library in production until its version 4 release.
 
-### Please note that the use of `hyp3lib` version 2 or less is **_NOT_** recommended. 
+The use of HyP3-lib version 2 or less is **_NOT_** recommended.
```
Comment (jhkennedy) [inline README.md:6]:
```suggestion
```

I don't think we need this since it's covered by the last line of the above alert

File: README.md [hunk 2]
```diff
@@ -1,14 +1,35 @@
...
 
-This library is outdated and largely untested, containing a broad collection of scripts/tools developed for upstream packages that have since undergone many *major* releases. 
-
-HyP3 plugins only use a subset of this repository, and version 3 will include drastic changes to reduce the library's scope to *just* those codes supporting HyP3 plugin development. We suggest *not* using this library, *especially* in a production environment, until version 3.0. 
+## HyP3-lib: A Common library for HyP3 plugins
```
Comment (jhkennedy) [inline README.md:8]: This should be line 1 of the README at a top level header (just `#`)... I've no idea why it was split into two lines before.

File: README.md [hunk 3]
```diff
@@ -1,14 +1,35 @@
...
+The goal of this package is to provide ASF's various HyP3 plugins with a set of utilities they can use to perform common tasks. This could include tasks such as downloading Sentinel-1 orbit data, or verifying that NASA EarthData credentials are available.
 
 If you're a MintPy user looking to crop HyP3 products, please see our [Time Series Analysis with HyP3 and Mintpy Tutorial](https://hyp3-docs.asf.alaska.edu/tutorials/mintpy).
 
-For other uses, we suggest checking out our [`asf-tools`](https://github.com/ASFHyP3/asf-tools) Python library and/or the [HyP3 Python SDK](https://github.com/ASFHyP3/hyp3-sdk). 
+If you are looking for a way to request HyP3 products via Python, we suggest checkout our [HyP3 Python SDK](https://github.com/ASFHyP3/hyp3-sdk).
+
+If you are looking for workflows that take advantage of our HyP3 on-demand products, we suggest checking out our [`asf-tools`](https://github.com/ASFHyP3/asf-tools) Python library.
```
Comment (jhkennedy) [inline README.md:15]:
```suggestion
If you are looking for workflows that take advantage of our HyP3 on-demand products, we suggest checking out our [ASF Tools](https://github.com/ASFHyP3/asf-tools) Python library.
```
I know this was here before, but it's more consistent this way I think. I can't decide if the link should include "Python library" or not...

File: README.md [hunk 4]
```diff
@@ -1,14 +1,35 @@
...
+## License
+The HyP3-lib package is licensed under the BSD 3-Clause license. See the LICENSE file for more details.
+
+## Code of conduct
+We strive to create a welcoming and inclusive community for all contributors to HyP3-lib. As such, all contributors to this project are expected to adhere to our code of conduct.
+
+Please see `CODE_OF_CONDUCT.md` for the full code of conduct text.
+
+## Contributing
+Contributions to the HyP3-lib library are welcome! If you would like to contribute, please submit a pull request on the GitHub repository.
```
Comment (jhkennedy) [inline README.md:26]:
```suggestion
Contributions to the HyP3-lib library are welcome! Whether you're finding bugs, adding new features, fixing anything broken, or improving documentation, get started by submitting an issue or pull request!

Please see [`CONTRIBUTING.md`](CONTRIBUTING.md) for more info.
```
I took the 2nd sentence from the contributing guide.

File: README.md [hunk 5]
```diff
@@ -1,14 +1,35 @@
...
+
+If you are looking for workflows that take advantage of our HyP3 on-demand products, we suggest checking out our [`asf-tools`](https://github.com/ASFHyP3/asf-tools) Python library.
+
+## License
+The HyP3-lib package is licensed under the BSD 3-Clause license. See the LICENSE file for more details.
+
+## Code of conduct
+We strive to create a welcoming and inclusive community for all contributors to HyP3-lib. As such, all contributors to this project are expected to adhere to our code of conduct.
+
+Please see `CODE_OF_CONDUCT.md` for the full code of conduct text.
```
Comment (jhkennedy) [inline README.md:23]:
```suggestion
Please see [`CODE_OF_CONDUCT.md`](CODE_OF_CONDUCT.md) for the full code of conduct text.
```

File: README.md [hunk 6]
```diff
@@ -1,14 +1,35 @@
...
 
 If you're a MintPy user looking to crop HyP3 products, please see our [Time Series Analysis with HyP3 and Mintpy Tutorial](https://hyp3-docs.asf.alaska.edu/tutorials/mintpy).
 
-For other uses, we suggest checking out our [`asf-tools`](https://github.com/ASFHyP3/asf-tools) Python library and/or the [HyP3 Python SDK](https://github.com/ASFHyP3/hyp3-sdk). 
+If you are looking for a way to request HyP3 products via Python, we suggest checkout our [HyP3 Python SDK](https://github.com/ASFHyP3/hyp3-sdk).
+
+If you are looking for workflows that take advantage of our HyP3 on-demand products, we suggest checking out our [`asf-tools`](https://github.com/ASFHyP3/asf-tools) Python library.
+
+## License
+The HyP3-lib package is licensed under the BSD 3-Clause license. See the LICENSE file for more details.
```
Comment (jhkennedy) [inline README.md:18]:
```suggestion
The HyP3-lib package is licensed under the BSD 3-Clause license. See the [`LICENSE`](LICENSE) file for more details.
```

File: README.md [hunk 7]
```diff
@@ -1,14 +1,35 @@
 # HyP3-lib
 
-## Common library for HyP3 plugins
+## CURRENTLY UNDER CONSTRUCTION!!!
+HyP3-lib is currently undergoing a major redesign and many utilities that are no longer used by ASF have been removed between version 2 and version 3. In addition, we plan to migrate some remaining functionality to other projects and to significantly refactor what remains. For this reason, **if you are using HyP3-lib we strongly recommend pinning to major versions at this time**. We further recommend not using this library in production until its version 4 release.
```
Comment (jhkennedy) [inline README.md:4]:
```suggestion
> [!WARNING]
> HyP3-lib is currently undergoing a major redesign and many utilities that are no longer used by existing HyP3 plugins have been removed between version 2 and version 3. In addition, we plan to migrate some remaining functionality to other projects and to significantly refactor what remains. For this reason, **if you are using HyP3-lib we strongly recommend pinning to major versions at this time**. We further recommend not using this library in production until its version 4 release.
```
I think using an alert here would pop better and be more appropriate
https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#alerts

**Review by jhkennedy — COMMENTED** [review]
Overall I really like it! Well done. 

I dropped a handful of small suggestions that I think are worth doing

**Comment by forrestfwilliams** [comment]
Thanks! @jhkennedy this is ready for a re-review!

**Review by jhkennedy — APPROVED** [review]

**Pull request merged**